src/inmanta/deploy/scheduler.py

```diff
@@ -672,7 +677,7 @@ async def report_resource_state(
                     # (or skip) causes it to become dirty now.
                     self._state.dirty.add(resource)
                 # propagate events
-                if details.attributes.get(const.RESOURCE_ATTRIBUTE_SEND_EVENTS, False):
+                if details.attributes.get(const.RESOURCE_ATTRIBUTE_SEND_EVENTS, False) and blocked_status != BlockedStatus.YES:
```

Comment:
Is this meant to check the target's blocked status instead? If so, I think @Hugo-Inmanta is incorperating that in his PR?

What we do need in this method is to set the resource's `BlockedStatus` depending on the deployment result -> either `NO` or `TRANSIENT`, but never overwriting `YES`.

Reply:
It was related to this comment:
> force event propagation only for transiently blocked resources, not YES blocked ones

Reply:
Operative word here is "force". That's about what Hugo is working on. I could've been more explicit on that.